apps/desktop-wallet/src/features/modals/modalSlice.ts

```diff
@@ -40,7 +41,7 @@ const modalSlice = createSlice({
         modalAdapter.removeOne(state, id)
       })
 
-    builder.addMatcher(isAnyOf(walletSwitched, walletLocked, activeWalletDeleted), (state) => {
+    builder.addMatcher(isAnyOf(walletSwitched, walletLocked, activeWalletDeleted, addressDeleted), (state) => {
```

Comment:
I understand why you did that here but I'm not sure I like it. The other actions (`walletSwitched`, `walletLocked` and `activeWalletDeleted`) are all at the wallet level. For those actions, it make sense to close all opened modal, because the whole context changes. 
 
It's different when you delete one address. I'm not sure that we should use this reducer every time we want to close more than 1 modal. In the future it could lead to closing modals that we don't want to close. WDYT?

One idea to fix this would be to allow `closeModal` to receive a modal name, allowing use to close specific modals from different part of the app.

Reply:
I acknowledge your concerns. 

It's true that you can get to the AddressOptionsModal (where the delete button is) in multiple ways:
- from the Addresses page -> AddressDetailsModal
- from the TransactionDetailsModal -> AddressDetailsModal
- from an infinite number of TransactionDetailsModal -> AddressDetailsModal -> TransactionDetailsModal -> ... -> 

in all above cases, deleting an address would close all modals. Which I think it's acceptable for now. But I agree it could be improved.

> One idea to fix this would be to allow closeModal to receive a modal name, allowing use to close specific modals from different part of the app.

That'd mean that ALL modals with the same name would be closed, right? Even if they are multiple. That's also a bit weird. I think we should look into another solution.

Maybe we shouldn't close any modals. Maybe we should just saw a snackbar that says "address deleted successfully." WDYT?

Reply:
Ah, now I remember why I wanted to close all modals. Imagine this:
1. User opens AddressDetailsModal
2. User opens AddressOptionsModal
3. User clicks "Forget"
4. User confirms in the ConfirmModal

If we only close the ConfirmModal and the AddressOptionsModal, then the AddressDetailsModal stays open with nothing to display

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/2a66cf0b-3f11-4d42-bef0-d89503651712">

What we would optimally want is to close ANY modals related to the deleted address. That can be a tricky thing to define. That's why I went for simply closing all modals for now.

How do you think we should proceed?

Reply:
If we want to identify which modals are related to an address, best I can think of is to create a reducer in the modal slice that loops over all open modals and checks if any of them includes a prop called `addressHash` that matches the deleted one...

Reply:
I would say that closing all modals with the same name would be a little better than closing all the modals (a bit more specific), but I agree that this isn't optimal either.
However, we could fix this by augmenting `closeModal` a bit, allowing to pass not only a name, but also a param to check? We could maybe try this with a new `closeModals` action:

Usage:
```ts
closeModals({ name: 'AddressDetailsModal', params: { addressHash: deletedAddressHash }) })
```

WDYT?

Reply:
Solution discussed on Slack and implemented :)